src/kitovu/sync/syncplugin.py

```diff
@@ -13,8 +13,16 @@ class AbstractSyncPlugin(metaclass=abc.ABCMeta):
     """
 
     @abc.abstractmethod
-    def connect(self, url: str, options: dict) -> None:
-        """Connect to the given URL.
+    def configure(self, info: typing.Dict[str, typing.Any]) -> None:
```

Comment:
I decided to separate setting a configuration in a plugin from connecting - that way, we can also do some basic error handling easily when things are missing in the config.